api/lib/infrastructure/repositories/schooling-registration-repository.js

```diff
@@ -147,63 +146,40 @@ module.exports = {
-    const [schoolingRegistrationsToUpdate, schoolingRegistrationsToCreate] = await this._getStudentsListToUpdateOrCreate(reconciledSchoolingRegistrationsToImport, existingSchoolingRegistrations);
+    const reconciledSchoolingRegistrationsToImport = await this._reconcileSchoolingRegistrations(schoolingRegistrationsFromFile, existingSchoolingRegistrations, domainTransaction);
 
     try {
-      await Promise.all([
-        bluebird.mapSeries(schoolingRegistrationsToUpdate, async (schoolingRegistrationToUpdate) => {
-          const attributesToUpdate = _.omit(schoolingRegistrationToUpdate, ['id', 'createdAt']);
-          const whereConditions = {
-            'organizationId': organizationId,
-            'nationalStudentId': schoolingRegistrationToUpdate.nationalStudentId,
-          };
-
-          await knexConn('schooling-registrations')
-            .where(whereConditions)
-            .update({
-              ...attributesToUpdate,
-              isDisabled: false,
-              updatedAt: knexConn.raw('CURRENT_TIMESTAMP'),
-            });
-        }),
-        knexConn.batchInsert('schooling-registrations', schoolingRegistrationsToCreate),
-      ]);
+      const schoolingRegistrationToSave = reconciledSchoolingRegistrationsToImport.map((schoolingRegistration) => ({
+        ..._.omit(schoolingRegistration, ['id', 'createdAt']),
+        updatedAt: knexConn.raw('CURRENT_TIMESTAMP'),
+        isDisabled: false,
+      }));
+
+      await knexConn('schooling-registrations')
+        .insert(schoolingRegistrationToSave)
+        .onConflict(['organizationId', 'nationalStudentId'])
+        .merge();
     } catch (err) {
       throw new SchoolingRegistrationsCouldNotBeSavedError();
     }
   },
 
-  async _reconcileSchoolingRegistrations(schoolingRegistrationsToImport, existingSchoolingRegistrations) {
+  async _reconcileSchoolingRegistrations(schoolingRegistrationsToImport, existingSchoolingRegistrations, domainTransaction) {
     const nationalStudentIdsFromFile = schoolingRegistrationsToImport.map((schoolingRegistrationData) => schoolingRegistrationData.nationalStudentId);
-    const students = await studentRepository.findReconciledStudentsByNationalStudentId(_.compact(nationalStudentIdsFromFile));
+    const students = await studentRepository.findReconciledStudentsByNationalStudentId(_.compact(nationalStudentIdsFromFile), domainTransaction);
 
-    return _.map(schoolingRegistrationsToImport, (schoolingRegistration) => {
-      const currentSchoolingRegistration = existingSchoolingRegistrations.find((currentSchoolingRegistration) => {
-        return currentSchoolingRegistration.nationalStudentId === schoolingRegistration.nationalStudentId;
+    _.each(students, (student) => {
+      const schoolingRegistration = schoolingRegistrationsToImport.find((schoolingRegistration) => student.nationalStudentId === schoolingRegistration.nationalStudentId);
```

Comment:
La réutilisation du nom `schoolingRegistration` pour deux variables distinctes ne me paraît pas très heureuse ici.
Suggestion : renommer la variable extérieure en `schoolingRegistrationToImport`.

Noter qu'on peut aussi s'éviter complètement le souci de nommage de la variable d'itération en écrivant :
```suggestion
      const schoolingRegistrationToImport = _.find(schoolingRegistrationsToImport, { nationalStudentId: student.nationalStudentId });
```

(applicable au `find` suivant également)